lib/charms/kubernetes_charm_libraries/v0/volumes.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Library used to leverage the Volumes Kubernetes in charms."""
+import logging
+from dataclasses import dataclass
+from typing import Iterable, Union
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.apps_v1 import StatefulSetSpec
+from lightkube.models.core_v1 import (
+    Container,
+    PodSpec,
+    PodTemplateSpec,
+    ResourceRequirements,
+    Volume,
+    VolumeMount,
+)
+from lightkube.resources.apps_v1 import StatefulSet
+from lightkube.resources.core_v1 import Pod
+from lightkube.types import PatchType
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class RequestedVolume:
+    """RequestedVolume."""
+
+    volume: Volume
+    volume_mount: VolumeMount
+
+
+class KubernetesRequestedVolumesError(Exception):
+    """KubernetesRequestedVolumesError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class KubernetesClient:
+    """Class containing all the Kubernetes specific calls."""
+
+    def __init__(self, namespace: str):
+        self.client = Client()
+        self.namespace = namespace
+
+    def pod_is_ready(
```

Comment:
I find the name of this method confusing because my first instinct to think to the status we see in a `kubectl get pods` command. not the fact that the addition of the volumes was successfully finished.